packages/core/src/stepFunctions/workflowStudio/workflowStudioEditorProvider.ts

```diff
@@ -115,6 +117,21 @@ export class WorkflowStudioEditorProvider implements vscode.CustomTextEditorProv
         _token: vscode.CancellationToken
     ): Promise<void> {
         await telemetry.stepfunctions_openWorkflowStudio.run(async () => {
+            // For invalid JSON, open default editor and show warning message
+            if (isInvalidJsonFile(document)) {
+                await vscode.commands.executeCommand('vscode.openWith', document.uri, 'default')
+                webviewPanel.dispose()
+                void vscode.window.showWarningMessage(i18n('AWS.stepFunctions.workflowStudio.actions.invalidJson'))
+
+                throw ToolkitError.chain(
+                    'Invalid JSON file',
+                    'The Workflow Studio editor was not opened because the JSON in the file is invalid',
+                    {
+                        code: 'InvalidJSONContent',
```

Comment:
`run()` will automatically set `reason` to this `code`. Thus there is no need for the `isInvalidJson` field.

Reply:
This code does not have anything to do with `isInvalidJson` metric, which indicates that the JSON code inside the custom webview is invalid. Meanwhile, this code is responsible for detecting invalid JSON code in local file to prevent opening webview integration in the first place.